thoth/storages/graph/postgres.py

```diff
@@ -4671,6 +4671,77 @@ def sync_dependency_monkey_result(self, document: dict) -> None:
 
                 inspection_run.dependency_monkey_run_id = dependency_monkey_run.id
 
+    def get_python_package_required_symbols(
+        self, package_name: str, package_version: str, index_url: str
+    ) -> List[str]:
+        """Get required symbols for a Python package in a specified version."""
+        package_name = self.normalize_python_package_name(package_name)
+        package_version = self.normalize_python_package_version(package_version)
+
+        with self._session_scope() as session:
+            query = session.query(PythonPackageVersionEntity)
+
+            query = (
+                query.join(
+                    HasArtifact,
+                    PythonPackageVersionEntity.id == HasArtifact.python_package_version_entity_id
+                )
+                .outerjoin(
+                    RequiresSymbol,
+                    RequiresSymbol.python_artifact_id == HasArtifact.python_artifact_id
+                )
+                .join(
+                    PythonPackageIndex,
+                    PythonPackageIndex.id == PythonPackageVersionEntity.python_package_index_id
+                )
+                .outerjoin(
+                    VersionedSymbol,
+                    RequiresSymbol.versioned_symbol_id == VersionedSymbol.id
+                )
+                .filter(PythonPackageVersionEntity.package_name == package_name)
+                .filter(PythonPackageVersion.package_version == package_version)
+                .filter(PythonPackageIndex.url == index_url)
+                .with_entities(VersionedSymbol.symbol)
+            )
+
+            # Query returns list of single tuples
+            # NOTE: can be empty even if request is valid
+            return [i[0] for i in query.all()]
+
+    def get_image_symbols(
```

Comment:
what about `get_analyzed_image_symbols`? and so adjust the description please.

Reply:
Based on the naming schema, should there be `_all` suffix?

Reply:
Correct, we get all now, not just one, thanks @fridex :)